src/sdk/bucketspace_fs.js

```diff
@@ -747,9 +765,10 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
         // If the system owner account wants to access the bucket, allow it
         if (is_system_owner) return true;
         const is_owner = (bucket.bucket_owner.unwrap() === account_identifier);
+        const is_same_root_account_owner = account.owner === bucket.owner_account.id; // case for IAM account
```

Comment:
how is bucket.owner_account an object with id field? isn't it just the id itself?

Reply:
You are probably right, I will check it.